app/src/main/java/eu/kanade/tachiyomi/ui/reader/PageIndicatorTextView.kt

```diff
@@ -42,20 +29,22 @@ class PageIndicatorTextView(
         val currText = " $text "
 
         // Also add a bit of spacing between each character, as the stroke overlaps them
-        val finalText = SpannableString(currText.asIterable().joinToString("\u00A0"))
+        val finalText = SpannableString(currText.asIterable().joinToString("\u00A0")).apply {
+            // Apply text outline
+            setSpan(spanOutline, 1, length-1, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
 
-        for (i in 1..finalText.lastIndex step 2) {
-            finalText.setSpan(ScaleXSpan(0.1f), i, i + 1, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+            for (i in 1..lastIndex step 2) {
+                setSpan(ScaleXSpan(0.3f), i, i + 1, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+            }
         }
-
-        super.setText(finalText, TextView.BufferType.SPANNABLE)
+        super.setText(finalText, BufferType.SPANNABLE)
     }
 
     private companion object {
-        // We need to use reflection to set the text color instead of using [setTextColor],
-        // otherwise the view is invalidated inside [onDraw] and there's an infinite loop
-        val textColorField = TextView::class.java.getDeclaredField("mCurTextColor").apply {
-            isAccessible = true
-        }!!
+        // A span object with text outlining properties
+        val spanOutline = OutlineSpan(
+                strokeColor = Color.rgb(45, 45, 45),
+                strokeWidth = 4f
+        )
```

Comment:
does the infinite loop not happen anymore?

Reply:
It doesn't, I tried setting a timber log in the onDraw method to check as well. Seems like setting the textColor in the onDraw has been resolved?